explorer/src/components/Mixnodes-DataGrid.tsx

```diff
@@ -0,0 +1,70 @@
+import { DataGrid, GridColDef, GridRenderCellParams } from '@mui/x-data-grid';
+import Button from '@mui/material/Button';
+import { Link } from 'react-router-dom';
+import { Box } from '@mui/system';
+import { useContext } from 'react';
+import { MainContext } from 'src/context/main';
+
+const columns = [
+  { field: 'owner', headerName: 'Owner', width: 500 },
+  {
+    field: 'identity_key',
+    headerName: 'Identity Key',
+    width: 500,
+    renderCell: (params: GridRenderCellParams) => {
+      return (
+        <Link to={`/network-components/mixnodes/${params.value}`} style={{ textDecoration: 'none', color: 'white', marginLeft: 16 }}>
+          {params.value}
+        </Link>
+      )
+    }
+  },
+  {
+    field: 'bond',
+    headerName: 'Bond',
+    width: 500,
+  },
+  {
+    field: 'host',
+    headerName: 'IP:Port',
+    width: 300,
+  },
+  {
+    field: 'location',
+    headerName: 'Location',
+    width: 150,
+  },
+  {
+    field: 'layer',
+    headerName: 'Layer',
+    width: 100,
+    type: 'number',
+  },
+];
+
+export const MixnodesDataGrid = () => {
+  const { mixnodes } = useContext(MainContext);
+  return (
+    <Box sx={{ height: 1080, width: '100%' }}>
+      <DataGrid
+        loading={mixnodes?.isLoading}
+        columns={columns}
+        rows={
+          mixnodes?.data?.map((m) => ({
+            id: m.owner,
+            owner: m.owner,
+            location: m.location?.country_name || '',
+            identity_key: m.mix_node.identity_key || '',
+            bond: m.bond_amount.amount || '',
+            host: m.mix_node.host || '',
+            layer: m.layer || '',
+          })) || []
+        }
+        pageSize={50}
+        rowsPerPageOptions={[5]}
+        disableSelectionOnClick
+      />
+    </Box>
+  );
+  return null;
+};
```

Comment:
Might be an idea to make this a component that accepts the data as a props. Can the be reused for Gateways?

Reply:
Absolutely. The intention originally was to make the MixnodesTable one table for reuse throughout app.
Now we have the datagrid looking good we can break it out further. 
Up to you and @mmsinclair really - if there's urgency to get a version of the app live quickly, deployed to staging etc - would suggest we park that for the mo (mixnodes table works). Then iterate and expand the DataGrid.

If we're doing okay for time, i'm happy to quickly sort that monday/tuesday. Won't take long.

Reply:
```            id: m.owner,
            owner: m.owner,
            location: m.location?.country_name || '',
            identity_key: m.mix_node.identity_key || '',
            bond: m.bond_amount.amount || '',
            host: m.mix_node.host || '',
            layer: m.layer || '',```

Seems odd to have to set empty strings everywhere. I think there are two options:
- a container component that does that mapping and then passes props down
- a hook that does it and watches the context state with an effect and sets its own state (to stop re-renders)

Reply:
I have since broken this out into a util function, takes the api response and returns a useable formatted array of objects for the data grid. Func is then used wherever the data grid is. But this will appear as part of the Blocks / Gateways work.

Currently looks like this:

```
export function mixnodeToGridRow(arrayOfMixnodes: MixNodeResponse) {
  
  let arr = [];
  
  arrayOfMixnodes !== undefined && arrayOfMixnodes.forEach((eachRecord: MixNodeResponseItem) => {
    let formattedRow = {
      id: eachRecord.owner,
      owner: eachRecord.owner,
      location: eachRecord?.location?.country_name || '',
      identity_key: eachRecord.mix_node.identity_key || '',
      bond: eachRecord.bond_amount.amount || '',
      host: eachRecord.mix_node.host || '',
      layer: eachRecord.layer || '',
    }
    arr.push(formattedRow);
  })
  return arr;
}
```

Then used as, follows... let me know if you'd prefer the hook you mentioned?

```
export const PageMixnodes: React.FC = () => {
  const { mixnodes } = useContext(MainContext);
  const [rows, setRows] = React.useState();

  React.useEffect(() => {
    if (mixnodes?.data !== undefined) {
      const formattedRows = mixnodeToGridRow(mixnodes.data);
      setRows(formattedRows);
    };
  }, [mixnodes])
  return (
    <>
      <Typography sx={{ marginBottom: 1 }} variant="h5">
        Mixnodes
      </Typography>
      <UniversalDataGrid
        loading={mixnodes?.isLoading}
        columnsData={columns}
        rows={rows}
        height={1080}
      />
    </>
  );
};
```